src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -674,6 +677,24 @@ public static float onAnvilRepair(Player player, @Nonnull ItemStack output, @Non
         MinecraftForge.EVENT_BUS.post(e);
         return e.getBreakChance();
     }
+    
+    public static boolean onGrindstoneChange(GrindstoneMenu container, @Nonnull ItemStack top, @Nonnull ItemStack bottem, Container outputSlot, int xp)
+    {
+        GrindstoneUpdateEvent e = new GrindstoneUpdateEvent(top, bottem);
+        if (MinecraftForge.EVENT_BUS.post(e))
+        {
+            xp = e.getXp();
+            return false;
+        }
+        if (e.getOutput().isEmpty()) 
+        {
+            return true;
+        }
+
+        outputSlot.setItem(0, e.getOutput());
+        xp = e.getXp();
```

Comment:
All assignments to the `xp` parameter are dropped when this method returns, they will never reach the `GrindstoneMenu#xp` variable you added.

Reply:
Oh that's stupid of me. The simpelst way to fix this is make use of an int[] or some "mutable int" instead. What would be the best way?

Reply:
I would say `int[]` or the `MutableInt` from apache-commons.

Reply:
Can you not return the int instead and use two negative numbers to represent true and false?

Reply:
Can you not return the int instead and use two negative numbers to represent true and false?

Reply:
Of course that also works! I'm really not paying attention.